front/components/assistant/conversation/AgentMessage.tsx

```diff
@@ -312,18 +312,18 @@ export function AgentMessage({
   >(null);
   useEffect(() => {
     if (
-      agentMessageToRender.action &&
-      isRetrievalActionType(agentMessageToRender.action) &&
-      agentMessageToRender.action.documents
+      agentMessageToRender.actions.length > 0 &&
```

Comment:
can we add a `deprecatedGetSingleActionFromAgentMessage` function (or whatever other name as long as it starts with deprecated) so it is easy to find and clean all of that stuff later 🙏